src/sema/contracts.rs

```diff
@@ -733,6 +733,39 @@ fn check_inheritance(contract_no: usize, ns: &mut ast::Namespace) {
     ns.diagnostics.extend(diagnostics);
 }
 
+/// Given a contract number, check for function names conflicting with any mangled name.
+/// Only applies to public functions.
+///
+/// Note: In sema we do not care about the function name too much.
+/// The mangled name is consumed later by the ABI generation.
+pub fn mangle_function_names(contract_no: usize, ns: &mut ast::Namespace) {
+    let public_functions: Vec<usize> = ns.contracts[contract_no]
+        .all_functions
+        .keys()
+        .copied()
+        .filter(|f| ns.functions[*f].is_public() && ns.functions[*f].ty == pt::FunctionTy::Function)
```

Comment:
total nitpick. put the `copied()` after the `filter()`

Reply:
Without it there would be a double deref `**f`. Besides it making for ugly code, I wouldn't bet on that having the better performance at all (`usize` is primitive)

Reply:
Fair point.